WalletWasabi/Wallets/Wallet.cs

```diff
@@ -568,4 +573,16 @@ public void UpdateUsedHdPubKeysLabels(Dictionary<HdPubKey, SmartLabel> hdPubKeys
 
 		KeyManager.ToFile();
 	}
+
+	private void LoadPrisonedCoinsState()
+	{
+		var prisonedCoinRecords = PrisonClient.PrisonedCoins;
+		foreach (var coin in Coins)
+		{
+			if (prisonedCoinRecords.FirstOrDefault(record => record.Outpoint.Equals(coin.Outpoint)) is { } record)
+			{
+				coin.BannedUntilUtc = record.BannedUntil;
+			}
+		}
+	}
```

Comment:
It's possible to load in a wallet's coins state, while in another wallet we are currently adding coins to the list or PrisonClient periodically replaces the list with a new instance. Very rare but could happen and cause problem?